jans-auth-server/server/src/main/java/io/jans/as/server/register/ws/rs/RegisterRestWebServiceImpl.java

```diff
@@ -311,6 +239,32 @@ private Response registerClientImpl(String requestParams, HttpServletRequest htt
                 throw errorResponseFactory.createWebApplicationException(Response.Status.BAD_REQUEST, RegisterErrorResponseType.INVALID_REDIRECT_URI, "Failed to validate redirect uris.");
             }
 
+            if (StringUtils.isNotBlank(r.getSubjectIdentifierAttribute())) {
+                if (!appConfiguration.getPublicSubjectIdentifierPerClientEnabled()) {
+                    throw errorResponseFactory.createWebApplicationException(
+                            Response.Status.BAD_REQUEST,
+                            RegisterErrorResponseType.INVALID_PUBLIC_SUBJECT_IDENTIFIER_ATTRIBUTE,
+                            "The public subject identifier per client is disabled."
+                    );
+                }
+
+                if (r.getSubjectType() != SubjectType.PUBLIC) {
+                    throw errorResponseFactory.createWebApplicationException(
+                            Response.Status.BAD_REQUEST,
+                            RegisterErrorResponseType.INVALID_PUBLIC_SUBJECT_IDENTIFIER_ATTRIBUTE,
+                            "The custom subject identifier requires public subject type."
+                    );
+                }
+
+                if (!appConfiguration.getSubjectIdentifiersPerClientSupported().contains(r.getSubjectIdentifierAttribute())) {
+                    throw errorResponseFactory.createWebApplicationException(
+                            Response.Status.BAD_REQUEST,
+                            RegisterErrorResponseType.INVALID_PUBLIC_SUBJECT_IDENTIFIER_ATTRIBUTE,
+                            "Invalid subject identifier attribute."
+                    );
+                }
+            }
+
```

Comment:
Sonar already complain about our methods lenght which is against "clean code", we should split it into methods. Please extract this snippet into separate method `validateSubjectIdentifierAttribute(r)`.

```
    public void validateSubjectIdentifierAttribute(RegisterRequest registerRequest) {
        if (StringUtils.isBlank(r.getSubjectIdentifierAttribute())) {
            return;
        }

        if (!appConfiguration.getPublicSubjectIdentifierPerClientEnabled()) {
            throw errorResponseFactory.createWebApplicationException(
                    Response.Status.BAD_REQUEST,
                    RegisterErrorResponseType.INVALID_PUBLIC_SUBJECT_IDENTIFIER_ATTRIBUTE,
                    "The public subject identifier per client is disabled."
            );
        }

        if (r.getSubjectType() != SubjectType.PUBLIC) {
            throw errorResponseFactory.createWebApplicationException(
                    Response.Status.BAD_REQUEST,
                    RegisterErrorResponseType.INVALID_PUBLIC_SUBJECT_IDENTIFIER_ATTRIBUTE,
                    "The custom subject identifier requires public subject type."
            );
        }

        if (!appConfiguration.getSubjectIdentifiersPerClientSupported().contains(r.getSubjectIdentifierAttribute())) {
            throw errorResponseFactory.createWebApplicationException(
                    Response.Status.BAD_REQUEST,
                    RegisterErrorResponseType.INVALID_PUBLIC_SUBJECT_IDENTIFIER_ATTRIBUTE,
                    "Invalid subject identifier attribute."
            );
        }
    }
```